### PR TITLE
Minor documentation overhaul

### PR DESCRIPTION
**Pull request by mbushkov**
* Updated Sphinx. Migrated from recommonmark to myst-parser. Removed a few hack from conf.py.
* Removed most mentions of AFF4 from the docs.
* Removed Python 2-specific sections.
* Fixed the docs to suggest to change mysql server config file to set max_allowed_packet to 40M (see https://github.com/google/grr/issues/947).
* Largely rewritten "Install from source": now telling how to build Fleetspeak wheels and specifically mentioning the case when we install from source to build a client template (see https://github.com/google/grr/issues/948).
* Made minor changes to "Build custom client templates".

You can see the version of the docs from this PR at https://grr-doc.readthedocs.io/en/sphinx-update/index.html.

### CODE REVIEW

File: deploying-grr-clients/index.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-* [Client Protection](protection.md)
-* [Troubleshooting ("I don't see my clients")](troubleshooting.md)
+```{toctree}
+---
+maxdepth: 1
+---
+Overview <overview.md>
+How to check if a deployed client talks back to GRR server <how-to-check-if-client-talks-back.md>
+On Windows <on-windows.md>
+On Mac OS X <on-mac-os-x.md>
```
Comment (max-vogler) [inline deploying-grr-clients/index.md:10]: Nit: Mac OS X is now called `macOS`

File: deploying-grr-clients/on-windows.md [hunk 2]
```diff
@@ -66,7 +66,7 @@ execute it. The installation doesn’t require user input.
 
 The expected output is something along these lines:
 
-```docker
+``docker
```
Comment (max-vogler) [inline deploying-grr-clients/on-windows.md:69]: qq: is this intended? Typically triple accent is used for code blocks. The same block is also closed with triple.

File: installing-grr-server/from-source.md [hunk 3]
```diff
@@ -1,114 +1,130 @@
...
-First, install the prerequisites:
-
-```bash
-sudo apt install -y fakeroot debhelper libffi-dev libssl-dev python3-dev \
-  python3-pip python3-venv wget openjdk-8-jdk zip git devscripts dh-systemd \
-  dh-virtualenv libc6-i386 lib32z1 asciidoc libmysqlclient-dev
-```
-
-Next, download the github repo and cd into its directory:
+Next, you need to build GRR. Download the github repo and cd into its directory:
```
Comment (max-vogler) [inline installing-grr-server/from-source.md:80]: Nit: `GitHub` instead of `github`.

**Review by max-vogler — APPROVED** [review]

File: deploying-grr-clients/index.md (hunk 1 above)
Comment (mbushkov) [reply deploying-grr-clients/index.md:10]: Fixed.

File: deploying-grr-clients/on-windows.md (hunk 2 above)
Comment (mbushkov) [reply deploying-grr-clients/on-windows.md:69]: That's accidental - thanks for spotting!

File: installing-grr-server/from-source.md (hunk 3 above)
Comment (mbushkov) [reply installing-grr-server/from-source.md:80]: Fixed.

**Pull request merged**